cmake/gtest_utils.cmake

```diff
@@ -0,0 +1,76 @@
+enable_testing()
+
+function(k2_add_gtest)
+  cmake_parse_arguments(
```

Comment:
This cmake code is very confusing, for me and I'm sure the majority of readers .    Can you please add some kind of comment that's aimed at someone who doesn't really understand cmake?  That explains what implications it has for usage, in terms of what flags you can pass or what behaviors it causes?

Reply:
Hmm, maybe I can change another way to reuse the gtest function. But for now, as to add a CMake module doesn't break things, I would like to put it here for convenience.